.github/workflows/validate_data.yml

```diff
@@ -0,0 +1,26 @@
+name: Validate supplied data against YAML schema
+
+on: [push,workflow_dispatch]
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+
+    - name: Setup python environment
+      shell: bash -el {0}
+      run: pip install .
+
+    - name: validate YAML data structure
+      shell: bash -el {0}
+      run: ls src/resolution_functions/instrument_data/*.yaml | xargs -n 1 python dev/validate_data_file.py --disable-resins
```

Comment:
If you are running on only one process, you could as well run all the yaml files with the script directly?

```
python dev/validate_data_file.py src/resolution_functions/instrument_data/*.yaml -dr
```

Reply:
Ok, have cleaned up a bit with Pathlib, simplified logic so that there is always a list of Path, and use default glob search to simplify the CI.